cirq-superstaq/cirq_superstaq/ops/qudit_gates.py

```diff
@@ -321,9 +357,16 @@ def _value_equality_values_(self) -> tuple[object, ...]:
         return (*super()._value_equality_values_(), self._dimension, self._level)
 
     def _value_equality_approximate_values_(self) -> tuple[object, ...]:
-        return (*super()._value_equality_approximate_values_(), self._dimension, self._level)
+        period = self._period()
+        if not period or cirq.is_parameterized(self._exponent):
+            exponent = self._exponent
+        else:
+            exponent = cirq.PeriodicValue(self._exponent, period)
+        return (exponent, self._global_shift, self._dimension, self._level)
```

Comment:
afaict everything works correctly if we just use
```suggestion
        return self._value_equality_values_()
```